### ISSUE TITLE
raw target: creating backups of multiple subvolumes to same target directory fails

### ISSUE THREAD

**Issue by Micha-Btz** [digint/btrbk#87]
hi there, i have some trouble to set up an raw target to my receiver. receiver locally mountet as nfs.

conf:

> 
> # Enable transaction log
> transaction_log            /var/log/btrbk.log
> 
> # creating snapshots for "rootfs" in "/mnt/vuplus" receiver nfs.
> volume /mnt/btrfs/root
> 	subvolume rootfs
> 	snapshot_dir             btrbk
> 	snapshot_create          always
> 	target raw /mnt/vuplus/sysiphus/btrbk    
> 	raw_target_compress  xz
> 	raw_target_encrypt   gpg
> 	gpg_keyring          /etc/btrbk/pubring.gpg
> 	gpg_recipient		mail
> 	incremental                yes

error message:

> btrbk --config=btrbk-vuplus.conf dryrun
> 
> ERROR: process died unexpectedly (btrbk v0.23.1)
> Please contact the author: Axel Burri <axel@tty0.ch>
> 
> Stack Trace:
> ----------------------------------------
> Use of uninitialized value in hash element at /usr/sbin/btrbk line 4278.
>  at /usr/sbin/btrbk line 219.
> 	main::__ANON__("Use of uninitialized value in hash element at /usr/sbin/btrbk"...) called at /usr/sbin/btrbk line 4278
> 

have i made a conf mistake? testet with 0.23 and 0.23.1 same error.

**Comment by digint** [comment]
> btrbk --config=btrbk-vuplus.conf dryrun
> 
> ERROR: process died unexpectedly (btrbk v0.23.1)
> Please contact the author: Axel Burri axel@tty0.ch

That's a bug. Can you reproduce the error? If yes, please provide me a full trace log (directly by email to keep some privacy):

```
btrbk -l trace dryrun  > /tmp/trace.log 2>&1
```

what would also help, a dump of:
```
btrbk config print-all > /tmp/config.out
```

**Comment by Micha-Btz** [comment]
the error is always the same if i try to write to the raw target.
i have a second conf write to an btrfs hdd per send receive target and this works fine.
mail is sent.

**Comment by digint** [comment]
Fixed in: 7326b9816c1c6a5001edbfa588d9698c3ff64845

This bug is triggered if backups of several subvolumes are created in the same target directory (which is perfectly legal).

**NOTE: This bug only affects raw targets. Normal (send-receive) backups are not affected by this.**

**Comment by Micha-Btz** [comment]
works well, thanks you for the fast response.
